### ISSUE TITLE
Wiki Update: get localkey with iPhone

### ISSUE THREAD

**Issue by JamesHabben** [pdugas/recteq#19]
I can't figure out how to update your wiki, but I have a couple things to add. Perhaps you can take this and update?

I don't have a rooted android device available to grab the local key. I was able to get the key with the free version of an application called iExplorer (https://macroplant.com/iexplorer). It isn't directly available, but here are steps that should reproduce what I did on a Windows computer. It might be different (easier) on a mac, and there is an iExplorer version for Mac. These steps assume you have an iPhone with the recteq app installed that has connected to your smoker. I did this on an iPhone 12 Pro (14.7.1) on a Windows 10 computer.

1. Ensure iTunes is installed and up to date
2. Plug in iPhone and use iTunes to create an encrypted backup (might be ok with non-encrypted but I didn't test that)
3. Install and open iExplorer
4. Scroll to the bottom in the left pane looking for "Browse iTunes Backups" and choose the name of the iPhone with the recteq app installed
5. Navigate to `Backup Explorer > App > com.rec.BarbecueMachine > Library > Preferences`
6. Select `com.rec.BarbecueMachine.plist`
7. In the right preview pane, scroll ~~about a third of the way down~~ almost to the bottom looking for `localKey`

![image](https://user-images.githubusercontent.com/6327329/127714414-7cae7eb6-bf82-47f5-9165-0b2350af900d.png)

**Comment by pdugas** [comment]
That should be handy for non-droid folk.  I'll work on getting this into the wiki soon.  Hip deep in other code atm.  Thanks much!

**Comment by ethan-holland** [comment]
I'm not able to find the localKey entry when looking at com.rec.BarbecueMachine.plist. Is it possible that this is out of date with the recent 2.0 app update? It looks like they may have changed the structure of what's stored in there.

**Comment by JamesHabben** [comment]
Hi @ethan-holland,
I just took a look, ~~and they did in fact move it. Its now inside `com.rec.BarbecueMachine` rather than `com.BarbecueMachine`~~. The `localKey` value is still in the same `com.rec.BarbecueMachine.plist` file, though its been moved lower down in the dataset.

edit: folder didn't change. I misread my notes. localKey is still in there.

**Comment by JamesHabben** [comment]
Here is a screen of the plist I looked at today. Notice the scroll bar is further down than the screen above. Might be easier to ctrl+a copy paste into notepad to search.
![image](https://user-images.githubusercontent.com/6327329/137385066-0290514a-324c-4cbe-a1cb-a80af54eb214.png)

**Comment by ethan-holland** [comment]
Interesting. There's something completely different about our files then. I'm using Atom (1.58) to view the plist file. When I do a search for "localKey" I get no results. I also searched for some of the other keys I see in your screenshot (isShare, verSW, gwid, etc.)  and they're not present either. So I checked the same file in Xcode and didn't find it there either. I can always go back to pairing the smoker to the Tuya app, but I'd love to be able to keep the smoker in the recteq app. Any other thoughts?

**Comment by JamesHabben** [comment]
Are you pulling this from an encrypted backup? Is that same plist there just with other values?

**Comment by ethan-holland** [comment]
Yes, it's an encrypted backup and yes it appears to be the same plist in that it's got the same name and appropriate sounding values for the name. What version of the recteq app are you running? I'm on 2.0.3.

**Comment by JamesHabben** [comment]
Well, this may be due to the version change. I've had the app on my phone for years. I just installed the current app an ipad, and I don't see the `localKey` value. My phone plist is 200kb while my ipad plist is only 4kb.
